extension/icu/icu-table-range.cpp

```diff
@@ -43,11 +43,33 @@ struct ICUTableRange {
 			if (U_FAILURE(success)) {
 				throw InternalException("Unable to create ICU calendar.");
 			}
+
+			timestamp_tz_t bounds[2];
+			interval_t step;
+			for (idx_t i = 0; i < inputs.size(); i++) {
+				if (inputs[i].IsNull()) {
+					return;
+				}
+				if (i >= 2) {
+					step = inputs[i].GetValue<interval_t>();
+				} else {
+					bounds[i] = inputs[i].GetValue<timestamp_tz_t>();
+				}
+			}
+			// Estimate cardinality using micros.
+			int64_t increment = 0;
+			if (!Interval::TryGetMicro(step, increment) || !increment) {
+				return;
+			}
+			const auto delta = bounds[1] - bounds[0];
```

Comment:
What happens if the subtraction leads to an overflow (e.g. `TIMESTAMP_MAX - TIMESTAMP_MIN`)? Can we add a test case for this?

Reply:
Oddly the ICU version doesn't need this because the binder checks the values to see if they can be represented by ICU... But I put it in anyway.